File: src/main/java/cloud/elb/BlocksActivated.scala
package cloud.elb

import se.sics.kompics.Event
import instance.common.Block
import java.util.{Collections, Set => JSet, HashSet => JHashSet, List => JList}

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class BlocksActivated(blks:JList[Block]) extends Event {

  val blocks: JSet[Block] = new JHashSet[Block]

  blocks.addAll(blks)

  def this() = this(Collections.emptyList())

  def addBlock(block:Block) = {
    blocks.add(block)
  }

}

File: src/main/java/cloud/elb/BlocksActivated.java
package cloud.elb;

import instance.common.Block;
import se.sics.kompics.Event;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-05-14
 *
 */

public class BlocksActivated extends Event {

	private Set<Block> blocks = new HashSet<Block>();
	
	public BlocksActivated() {
		
	}
	
	public BlocksActivated(List<Block> blocks) {
		this.blocks.addAll(blocks);
	}
	
	public void addBlock(Block block) {
		this.blocks.add(block);
	}

	public Set<Block> getBlocks() {
		return blocks;
	}

}
